core/interfaces/i_registrable_field.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a Blockly field that can be registered.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IRegistrableField');
+
+goog.requireType('Blockly.Field');
+
+/**
+ * A registrable field.
```

Comment:
Add a comment somewhere in here explaining why we couldn't make it an interface.

Reply:
Done. Please check if that makes sense.